### DISCUSSION TITLE
RecordBuilder Enhancer

### DISCUSSION THREAD

**Discussion by Randgalt** [Randgalt/record-builder#120] [Show and tell]
https://github.com/Randgalt/record-builder/pull/119

**FOR YOUR CONSIDERATION**

This is an idea that I've had for a while: being able to inject null checks and defensive copying into Java Record default constructors. While working on the idea I was able to generalize it into something that can be customizable as part of a normal build process. 

See the [README](https://github.com/Randgalt/record-builder/blob/jordanz/enhancer/record-builder-enhancer/README.md) for complete details.

I haven't decided on whether or not to release this yet. If people find it useful I will. Use `<version>34-SNAPSHOT</version>` to test it. 

**THANK YOU**

**Comment by rlei** [comment]
Love this and absolutely useful! Please do consider release the enhancer with a stable version number 😃
